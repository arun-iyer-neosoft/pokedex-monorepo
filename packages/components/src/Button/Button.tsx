import React from 'react';

interface ButtonProps extends React.HTMLProps<HTMLButtonElement> {
	variant?: 'light' | 'primary';
	type?: 'button' | 'submit' | 'reset';
	isLoading?: boolean;
	children: React.ReactNode;
}

export const Button = ({
	variant,
	type,
	children,
	isLoading,
	disabled,
	...rest
}: ButtonProps) => {
	const colors =
		variant === 'light'
			? 'bg-gray-200 hover:bg-gray-300 text-black'
			: 'bg-blue-600 hover:bg-blue-700 text-white';
	return (
		<button
			disabled={disabled || isLoading}
			className={`mb-12 px-6 py-2 rounded-lg text-xl ${colors} disabled:opacity-70`}
			type={type ?? 'button'}
			{...rest}
		>
			{children}
		</button>
	);
};
