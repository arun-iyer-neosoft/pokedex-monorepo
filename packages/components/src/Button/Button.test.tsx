import { render, screen } from '@testing-library/react';
import * as React from 'react';
import { Button } from './Button';

describe('Button', () => {
	it('Renders the Button with text', () => {
		const { container } = render(<Button>click me</Button>);

		expect(screen.getByText('click me')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('Renders button with the correct variant color', () => {
		const { container } = render(<Button variant='primary'>click me</Button>);

		expect(screen.getByText('click me').closest('button')).toHaveClass(
			'bg-blue-600'
		);
		expect(container).toMatchSnapshot();
	});

	it('Button should be disabled when loading is true', () => {
		const { container } = render(<Button isLoading={true}>click me</Button>);

		expect(screen.getByText('click me').closest('button')).toBeDisabled();
		expect(container).toMatchSnapshot();
	});
});
