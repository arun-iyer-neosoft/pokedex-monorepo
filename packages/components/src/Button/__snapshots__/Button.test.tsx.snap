// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button should be disabled when loading is true 1`] = `
<div>
  <button
    class="mb-12 px-6 py-2 rounded-lg text-xl bg-blue-600 hover:bg-blue-700 text-white disabled:opacity-70"
    disabled=""
    type="button"
  >
    click me
  </button>
</div>
`;

exports[`Button Renders button with the correct variant color 1`] = `
<div>
  <button
    class="mb-12 px-6 py-2 rounded-lg text-xl bg-blue-600 hover:bg-blue-700 text-white disabled:opacity-70"
    type="button"
  >
    click me
  </button>
</div>
`;

exports[`Button Renders the Button with text 1`] = `
<div>
  <button
    class="mb-12 px-6 py-2 rounded-lg text-xl bg-blue-600 hover:bg-blue-700 text-white disabled:opacity-70"
    type="button"
  >
    click me
  </button>
</div>
`;
